### PR TITLE
Add option to hide past events.

### LINKED ISSUE #1: Hide completed events
I know this might be out of the scope of this extension but do you know of a way to completely hide completed events?

### PR DESCRIPTION
**Pull request by jekkel**
Fixes #1

### CODE REVIEW

File: src/patch/new-reload-events.ts [hunk 1]
```diff
@@ -25,14 +25,30 @@ export function buildPatchedReloadEventsFunction(
...
+      if (styleAsCompleted && patchConfiguration.shouldHidePastEvents) {
+        continue;
+      }
+
+      let style = null;
+      if (styleAsCompleted) {
+        style = getCompletedEventStyle();
+      } else if (styleAsOngoing) {
+        style = getOngoingEventStyle();
+      }
```
Comment (marcinjahn) [inline src/patch/new-reload-events.ts:46]:
```suggestion
      let style = styleAsCompleted 
        ? getCompletedEventStyle() 
        : styleAsOngoing 
          ? getOngoingEventStyle() 
          : null;
```

File: src/patch/new-reload-events.ts [hunk 2]
```diff
@@ -60,7 +76,10 @@ export function buildPatchedReloadEventsFunction(
   };
 }
 
-function isCompletedEvent(event: Event, shouldStylePastDays: boolean) {
+function shouldBeStyledAsCompletedEvent(
```
Comment (marcinjahn) [inline src/patch/new-reload-events.ts:79]:
```suggestion
function isCompletedEvent(
```

File: src/patch/new-reload-events.ts [hunk 3]
```diff
@@ -74,7 +93,10 @@ function isCompletedEvent(event: Event, shouldStylePastDays: boolean) {
   return isFinished && !startedInSomePastDay;
 }
 
-function isOngoingEvent(event: Event, shouldStyleOngoingEvents: boolean) {
+function shouldBeStyledAsOngoingEvent(
```
Comment (marcinjahn) [inline src/patch/new-reload-events.ts:96]:
```suggestion
function isOngoingEvent(
```

File: src/patch/new-reload-events.ts (hunk 1 above)
Comment (marcinjahn) [inline src/patch/new-reload-events.ts:46]:
```suggestion
      const style = styleAsCompleted 
        ? getCompletedEventStyle() 
        : styleAsOngoing 
          ? getOngoingEventStyle() 
          : null;
```

**Review by marcinjahn — APPROVED** [review]
LGTM, thank you for your contribution, solid work!
The one thing that I guess might be a little bit confusing is that you have to toggle both Past Events toggles to get the hiding behavior. Normally, the UI should not allow you to set things that are invalid, and you'll be able to do so in the current shape of it.
However, I'm still gonna accept it, since I'm too lazy to change it myself :P and I'm not going to  block your PR.

File: src/patch/new-reload-events.ts (hunk 1 above)
Comment (jekkel) [reply src/patch/new-reload-events.ts]: Legit style preference :)

Personally, I think multiple ternary statements on multiple lines are too hard to read for almost no benefit.

**Comment by jekkel** [comment]
Thanks for merging this! I don't really get what you meant though. From my POV the setting hidden to true is compatible with both style past day events true and false. Granted, if both are true, almost all past events are hidden and I can't really think of a use case for it, but anyway. I'm with you here - given the simplicity of the overall thing it's likely not worth the trouble to make better than that :)

**Pull request merged**